Minimap

computer-kyle opened the pull request:
## Description

Adds a minimap that shows a 2D, overhead representation of game objects

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [x] Enhancement
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please specify):

## Related Issue

N / A

## Changes Made

- Adds a minimap that shows the location of both the player and antagonist using color-coded dots
- Minimap can be displayed by pressing and holding the [M] key
- The locations of minigames are displayed on the map
- Code is present that changes the sprite for each minimap based on its completion status

## Checklist

- [x] Code follows the project's coding style.
- [x] If asset(s) added, they have been made into a prefab
- [ ] Tests have been added/updated.
- [x] Documentation has been updated (if applicable).
- [x] No unused code or assets remain.
- [x] Unity project builds without errors.

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/cf8e809e-5581-4462-864e-022c0d0440cc)

## Additional Notes

The minimap has been added as a prefab in the Prefabs/Overlays directory. It should be added to Overlays in Scene, with the Player and NewAntagonist transform added to the script in the editor to function

## Relevant UML Diagrams
![image](https://github.com/user-attachments/assets/38d2bb33-5e7a-490b-ac72-e7ab7982832f)

computer-kyle committed:
Add Map Assets & Create Script

- Adds map assets: Map and Background
- Adds these elements to Overlay
- Adds a Map Controller script

computer-kyle committed:
Testing ways to draw points to screen

computer-kyle committed:
Added dot asset for player, improved MapController.cs script

- Added a dot which will act as the asset that indicates the Player and Antagonist (with different colors)

- Updated script to accurately show player on the minimap

computer-kyle committed:
Added 2 dot assets for Antagonist and Player, colored

- Changed from one dot asset to two, colored based on the GameObject they represent

computer-kyle committed:
Added Antagonist to MapController.cs

computer-kyle committed:
Updated MapController.cs to only work with keypress

- Added code that checks for a key [M] being pressed to display the map and run the calculations to translate positions to display coordinates

computer-kyle committed:
Prefabbed Map, removed from Scene

- Added Map and children to a prefab
- Removed from SampleScene to avoid merge conflicts

computer-kyle committed:
Added ghost "floating" to NewAntagonistController.cs

- Added a line of code that updates the antagonist's y-position based on a Sin function taking the game Time as an input, with modifiers

computer-kyle committed:
Add documentation

- Adds documentation to MapController.cs

computer-kyle committed:
Merge branch 'main' into minimap

krain0 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - **Adds minimap:** The minmap looks good and accurately tracks the location of the player and antagonist. It makes the game a bit easier, which I think is good because before it was easy to lose track of where you were
  - **Activated by pressing M:** Pressing M consistently activates the map
  - **Ghost floating animation:** This seems to work as intended, and it looks great

- **Unused Code**:
  - None

- **Code Style**:
  - Fits with our style

- **Script Structure**:
  - All new assets were placed in reasonable folders.

- **Documentation**:
  - The comments provide an appropriate amount of explanation

- **Build Integrity**:
  - The game built without issue on my machine.

---

### Summary
I think the minimap was a great addition to help players navigate the map. It is well implemented, and fits with the visual style of the game. Activating/deactivating it works without issues. I also enjoy the floating animation that you added to the antagonist. It makes the movement look much more natural. Looks ready to merge.

brandon722 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - **Adds minimap:** This minimap looks great! I like the detail of the ghost floating animation. I agree with Kyle's PR review where when the player is first playing it is easy for them to get lost, so this minimap is a great addition.

- **Unused Code**:
  - None

- **Code Style**:
  - Aligns with our style and documentation makes the code easy to understand. 

- **Script Structure**:
  - All new assets were placed in reasonable folders.

- **Documentation**:
  - The comments explain the code clearly.

- **Build Integrity**:
  - The game builds with no errors.

---

### Summary
This was a really good idea, and it looks great! This is an awesome addition, it works perfectly; ready to merge.

computer-kyle committed:
Moved assets for Minimap, added assets

- Moved existing assets for Minimap to resource folder, so they can be easily referenced programatically

computer-kyle committed:
Update Map.prefab

computer-kyle committed:
Update Map.prefab to include placeholder for minigame UI element

computer-kyle committed:
Updated Minigames code, minimap now displays location and completion status of minigames

computer-kyle committed:
Comment out code that updates sprites when minigames are completed
